packages/material-ui/src/Select/SelectInput.js

```diff
@@ -95,6 +96,14 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
     [value],
   );
 
+  React.useEffect(() => {
+    return () => {
+      if (handleClick.current) {
+        document.removeEventListener('click', handleClick.current, true);
```

Comment:
How do we know on which document we added the listener? Below you remove it from a specific document. Here you remove it from the current document in the scope.

Reply:
Fixed.